synapse/handlers/_base.py

```diff
@@ -57,10 +57,24 @@ def ratelimit(self, requester):
         time_now = self.clock.time()
         user_id = requester.user.to_string()
 
+        # Disable rate limiting of users belonging to any AS that is configured
+        # not to be rate limited in its registration file (rate_limited: true|false).
+        # The AS user itself is never rate limited.
+
         app_service = self.store.get_app_service_by_user_id(user_id)
         if app_service is not None:
             return  # do not ratelimit app service senders
 
+        should_rate_limit = True
+
+        for service in self.store.get_app_services():
+            if service.is_interested_in_user(user_id):
+                should_rate_limit = service.is_rate_limited()
+                break
```

Comment:
You can probably compact the lot of this and remove `should_rate_limit` entirely:
```py
for service in self.store.get_app_services():
    if service.is_interested_in_user(user_id) and not service.is_rate_limited():
        return
```

But I defer to Erik's view on whether this is clearer / what he would prefer.